### PR TITLE
Change HIVE_WRITER_ERROR to be more specific

### PR DESCRIPTION
**Pull request by raghavsethi**
We now break down the error codes into more specific categories -
HIVE_WRITER_DATA_ERROR and HIVE_WRITER_METADATA_ERROR to better
understand what fraction of errors come from the metastore vs HDFS.

### CODE REVIEW

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveErrorCode.java [hunk 1]
```diff
@@ -37,14 +37,15 @@
...
-    HIVE_SERDE_NOT_FOUND(0x0100_0012),
-    HIVE_UNSUPPORTED_FORMAT(0x0100_0013),
-    HIVE_PARTITION_READ_ONLY(0x0100_00014),
-    HIVE_TOO_MANY_OPEN_PARTITIONS(0x0000_0015),
-    HIVE_CONCURRENT_MODIFICATION_DETECTED(0x0000_0016),
-    HIVE_COLUMN_ORDER_MISMATCH(0x0000_0017),
-    HIVE_FILE_MISSING_COLUMN_NAMES(0x0000_0018);
+    HIVE_WRITER_DATA_ERROR(0x0100_0011),
+    HIVE_WRITER_METADATA_ERROR(0x0100_0012),
+    HIVE_SERDE_NOT_FOUND(0x0100_0013),
```
Comment (dain) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveErrorCode.java]: Why renumber this error code

**Comment by dain** [comment]
Looks good

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveErrorCode.java [hunk 2]
```diff
@@ -37,14 +37,16 @@
...
+    HIVE_DATA_WRITE_ERROR(0x0100_0011),
     HIVE_SERDE_NOT_FOUND(0x0100_0012),
     HIVE_UNSUPPORTED_FORMAT(0x0100_0013),
     HIVE_PARTITION_READ_ONLY(0x0100_00014),
     HIVE_TOO_MANY_OPEN_PARTITIONS(0x0000_0015),
     HIVE_CONCURRENT_MODIFICATION_DETECTED(0x0000_0016),
     HIVE_COLUMN_ORDER_MISMATCH(0x0000_0017),
-    HIVE_FILE_MISSING_COLUMN_NAMES(0x0000_0018);
+    HIVE_FILE_MISSING_COLUMN_NAMES(0x0000_0018),
+    HIVE_FILE_CLOSE_ERROR(0x0100_0019),
```
Comment (electrum) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveErrorCode.java]: `HIVE_CLOSE_WRITER_ERROR`

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveErrorCode.java [hunk 3]
```diff
@@ -37,14 +37,16 @@
     HIVE_DATABASE_LOCATION_ERROR(0x0100_000E),
     HIVE_PATH_ALREADY_EXISTS(0x0100_000F),
     HIVE_FILESYSTEM_ERROR(0x0100_0010),
-    HIVE_WRITER_ERROR(0x0100_0011),
```
Comment (electrum) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveErrorCode.java]: Let's just delete this one so we don't reuse the error code

Comment (electrum) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveErrorCode.java]: Replace the line with a comment
```java
// code 0x0100_0011 is deprecated

**Comment by raghavsethi** [comment]
@electrum ready for review

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveErrorCode.java [hunk 4]
```diff
@@ -37,14 +37,17 @@
...
+//  code HIVE_WRITER_ERROR(0x0100_0011), is deprecated
```
Comment (haozhun) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveErrorCode.java]: remove comma

**Pull request merged**